### PR TITLE
Fix a bug in addData and subData functions

### PR DESCRIPTION
**Pull request by evillique**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 7643e216bead8c98d742427869cd121655cbdd28 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/54636/7643e216bead8c98d742427869cd121655cbdd28/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>✅ success</td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>⏳ pending</td></tr>
<tbody>
</table>

File: src/Functions/FunctionsOpDate.cpp [hunk 1]
```diff
@@ -53,7 +53,6 @@ class FunctionOpDate : public IFunction
     }
 
     bool useDefaultImplementationForConstants() const override { return true; }
-    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {0, 2}; }
```
Comment (novikd) [inline src/Functions/FunctionsOpDate.cpp:56]: Why did you remove 2, but didn't add any test for that?

Comment (evillique) [reply src/Functions/FunctionsOpDate.cpp:56]: Because `2` is a 3rd argument and we only accept 2

**Review by novikd — APPROVED** [review]
